crates/python/src/py_sync.rs

```diff
@@ -0,0 +1,106 @@
+/// Spawn and block on a future using the pyo3 tokio runtime.
+/// Useful for returning a synchronous `PyResult`.
+///
+///
+/// When used like the following:
+/// ```rs
+/// async fn say_hello(name: String) -> String {
+///     format!("hello {name}")
+/// }
+///
+/// #[pyo3(name="say_hello")]
+/// pub fn py_say_hello(name: String) -> PyResult<String> {
+///     py_sync!(say_hello(name))
+/// }
+/// ```
+///
+/// Becomes the associated "synchronous" python call:
+/// ```py
+/// assert say_hello("Rigetti") == "hello Rigetti"
+/// ```
+macro_rules! py_sync {
+    ($body: expr) => {{
+        let runtime = pyo3_asyncio::tokio::get_runtime();
+        let handle = runtime.spawn($body);
+        runtime
+            .block_on(handle)
+            .map_err(|err| PyRuntimeError::new_err(err.to_string()))?
+    }};
+}
+
+/// Convert a rust future into a Python awaitable using
+/// `pyo3_asyncio::tokio::local_future_into_py`
+macro_rules! py_async {
```

Comment:
Thoughts on naming this `py_local_async` and have a `py_async` using `future_into_py`?

Reply:
Just changed the `py_async` to use `future_into_py` which works great, don't know why I was using `local_future_into_py` but it's unnecessary. Good catch!

Reply:
The difference IIRC is that `future_into_py` requires the `Future` to be either `Send` or `Sync` (or both), while `local_future_into_py` doesn't have the same restriction. I think it's the difference of whether the future can be run on a different thread or not.

I think it's good to have the option to use `local_future_into_py`, but that can also be added when needed.